### ISSUE TITLE
Can't parse type parameters containing trailing +

### ISSUE THREAD

**Issue by Arnavion** [dtolnay/syn#120]
```rust
let raw = r#"
    fn main() {
        let x: Box<Debug+>;
    }
"#;

let actual = parse_item(raw).unwrap(); // Fails
```

Found in [`tests/rust/src/test/ui/token/trailing-plus-in-bounds.rs`](https://github.com/rust-lang/rust/blob/ddc5d7bd4b9ea3e8a8ccf82cb443e950be311d61/src/test/ui/token/trailing-plus-in-bounds.rs#L16) which should pass without the trailing `FAIL`

**Comment by dtolnay** [comment]
Wow I had no idea this was allowed!

**Comment by mystor** [comment]
Shouldn't be hard so putting it in the 0.12 milestone.

**Comment by dtolnay** [comment]
Fixed in #265.

**Comment by eddyb** [comment]
This was mistakenly fixed only as a special case of a general argument, but [it's valid syntax for any `dyn`/`impl` (including old `dyn`-less trait objects)](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4a62af0321276bd9ca9af4911dfbf53d).

Filing a new issue now.
